lib/https.js

```diff
@@ -572,30 +572,34 @@ Agent.prototype._cacheSession = function _cacheSession(key, session) {
   if (this.maxCachedSessions === 0)
     return;
 
+  // Cache property accesses for better performance
+  const { map, list } = this._sessionCache;
+
   // Fast case - update existing entry
-  if (this._sessionCache.map[key]) {
-    this._sessionCache.map[key] = session;
+  if (map[key]) {
+    map[key] = session;
     return;
   }
 
   // Put new entry
-  if (this._sessionCache.list.length >= this.maxCachedSessions) {
-    const oldKey = ArrayPrototypeShift(this._sessionCache.list);
+  if (list.length >= this.maxCachedSessions) {
+    const oldKey = ArrayPrototypeShift(list);
     debug('evicting %j', oldKey);
-    delete this._sessionCache.map[oldKey];
+    delete map[oldKey];
   }
 
-  ArrayPrototypePush(this._sessionCache.list, key);
-  this._sessionCache.map[key] = session;
+  ArrayPrototypePush(list, key);
+  map[key] = session;
 };
 
 Agent.prototype._evictSession = function _evictSession(key) {
-  const index = ArrayPrototypeIndexOf(this._sessionCache.list, key);
+  const { map, list } = this._sessionCache;
```

Comment:
There is no need to optimize `map`, as it's only used once. I would also avoid destructuring.